libs/wingc/src/diagnostic.rs

```diff
@@ -111,6 +111,22 @@ impl WingSpan {
 			false
 		}
 	}
+
+	/// Returns the source code text of the span (reads the file from disk and extracts the text)
+	/// (currently only handles single line spans)
+	pub fn code(&self) -> String {
+		let Ok(contents) = fs::read_to_string(&self.file_id) else {
```

Comment:
I think we might we can improve the performance of this in the future by storing a byte offset into the original file inside `Position`, so that we don't have to read the entire file when we are obtaining the span. Add a TODO / issue?

(We should probably avoid loading entire source files into memory multiple times while compiling source code)

Reply:
Yes, agreed. I suspect this will be primarily used for debugging purposes. I am wondering if there's a way to make this clearer from the name or some Rust annotation?

Reply:
You could try using `pub(crate)` to say it's only meant to be used internally, or just add a comment